### ISSUE TITLE
Define @parameter.outer which would include punctuation

### ISSUE THREAD

**Issue by andreypopp** [nvim-treesitter/nvim-treesitter-textobjects#22]
First of all, thanks for this plugin! I'm trying it out and the results are impressive!

One thing I'm missing from [targets.vim](https://github.com/wellle/targets.vim#an-argument) is the "an argument" text object:

> Select an argument in a list of arguments. Includes a separator if preset, but excludes surrounding braces. This leaves a proper argument list after deletion. Accepts a count.
> ```
>       ...........
> a , b ( cccccccc , d ) e
>         └─── aa ──┘
> ```

As I understand the analogue could be `@parameter.outer`. Unfortunately I wasn't able to see how that could be defined with queries at the moment as the most grammars I've checked do not keep punctuation in the AST. Maybe some out of band solution might work where there's some post processing on the query result to find the matching comma...

**Comment by theHamsta** [comment]
We didn't implement it yet because of the limitation that a query could only yield a single tree-sitter node. We've recently added a new way to add multinode ranges. You would specify the start node and the end node. E.g.

```
(( 
  [(method_signature) (function_signature)] @_start
  .
  (function_body) @_end
)
(make-range! "function.outer" @_start @_end))

```

So `@parameter.outer` should now be feasible to implement.

**Comment by theHamsta** [comment]
Fixed via #23 . Well, only for Rust, C++ and Python...
